zeeguu_api/api/dashboard.py

```diff
@@ -0,0 +1,297 @@
+    except ValueError:
+        flask.abort(400)
+
+
+# Removes class.Can only be called successfuly if the class is empty. Otherwise we return 400 (Value error).
+@api.route("/remove_class/<class_id>", methods=["POST"])
+@with_session
+def remove_class(class_id):
+    from zeeguu.model import TeacherCohortMap
+    if (not _has_permission_for_class(class_id)):
+        flask.abort(401)
+    try:
+        selected_cohort = Cohort.query.filter_by(id=class_id).one()
+
+        if not selected_cohort.cur_students == 0:
+            flask.abort(400)
+
+        links = TeacherCohortMap.query.filter_by(cohort_id=class_id).all()
+        for link in links:
+            db.session.delete(link)
+        db.session.delete(selected_cohort)
+        db.session.commit()
+        return 'removed'
+    except ValueError:
+        flask.abort(400)
+
+
+# Takes Teacher id as input and outputs list of all cohort_ids that teacher owns
+@api.route("/get_classes", methods=["GET"])
+@with_session
+def get_classes_by_teacher_id():
+    from zeeguu.model import TeacherCohortMap
+    mappings = TeacherCohortMap.query.filter_by(user_id=flask.g.user.id).all()
+    cohorts = []
+    for m in mappings:
+        info = _get_class_info(m.cohort_id)
+        cohorts.append(info)
+    return json.dumps(cohorts)
+
+
+# Takes cohort_id and reuturns dictionary with relevant class variables
+@api.route("/get_class_info/<id>", methods=["GET"])
+@with_session
+def wrapper_to_json_class(id):
+    if (not _has_permission_for_class(id)):
+        flask.abort(401)
+    return jsonify(_get_class_info(id))
+
+
+def _get_class_info(id):
+    c = Cohort.find(id)
+    class_name = c.class_name
+    inv_code = c.inv_code
+    max_students = c.max_students
+    cur_students = c.cur_students
+    class_language_id = c.class_language_id
+    class_language = Language.query.filter_by(id=class_language_id).one()
+    d = {'id': str(id), 'class_name': class_name, 'inv_code': inv_code, 'max_students': max_students,
+         'cur_students': cur_students, 'class_language_name': class_language.name, 'class_id': id}
+    return d
+
+
+# Takes two inputs (user_id, cohort_id) and links them to one another in teacher_cohort_map table.
+def _link_teacher_class(user_id, cohort_id):
+    from zeeguu.model import TeacherCohortMap
+    user = User.find_by_id(user_id)
+    cohort = Cohort.find(cohort_id)
+    db.session.add(TeacherCohortMap(user, cohort))
+    db.session.commit()
+    return 'added teacher_class relationship'
+
+
+# Checks if the inputted invite code is already in use.
+@api.route("/check_invite_code/<invite_code>", methods=["GET"])
+@with_session
+def check_inv_code(invite_code):
+    c = Cohort.query.filter_by(inv_code=invite_code).first()
+    if c is None:
+        return jsonify(1)
+    return jsonify(0)
+
+
+# creates a class in the data base. Requires form input (inv_code, class_name, class_language_id, max_students, teacher_id)
+@api.route("/add_class", methods=["POST"])
```

Comment:
`/add_own_cohort` would make it even more explicit that 
the cohort is added to the logged in user. 

It might be possible that in the future, another user, a super user
or admin would be able to `/create_cohort` for a teacher. thus, 
the `own` would make a difference.